ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java

```diff
@@ -186,20 +187,32 @@ public UnresolvedPlan visitWhereCommand(WhereCommandContext ctx) {
   @Override
   public UnresolvedPlan visitJoinCommand(OpenSearchPPLParser.JoinCommandContext ctx) {
     Join.JoinType joinType = getJoinType(ctx.joinType());
-    if (ctx.joinCriteria() == null) {
-      joinType = Join.JoinType.CROSS;
+    List<Argument> arguments =
+        ctx.joinOption().stream().map(o -> (Argument) expressionBuilder.visit(o)).toList();
+    Argument.ArgumentMap argumentMap = Argument.ArgumentMap.of(arguments);
+    if (argumentMap.get("type") != null) {
+      Join.JoinType joinTypeFromArgument = getJoinType(argumentMap.get("type").toString());
+      if (ctx.joinType() == null) {
```

Comment:
JoinType is not a valid token in SPL mode. The AST builder should be aware of SPL mode and handle it separately.